components/compliance-service/api/tests/test_ingest_to_mgr_conn.rb

```diff
@@ -20,6 +20,12 @@
     nodes_list = MANAGER_GRPC nodes, :list, Nodes::Query.new()
     assert_equal(nodes_list.total, 9)
 
+    nodes_list['nodes'].each { |node|
+      assert_equal(node['scan_data'].id.length > 0, true)
+      assert_equal(node['scan_data'].status.length > 0, true)
+      assert_equal(node['scan_data'].end_time.length > 0, true)
+    }
```

Comment:
If this ever failed it'd give a pretty opaque error message. If you rewrote it a bit it could give a lot more information about what went wrong:

```
Petes-MBP:tmp pete$ cat test.rb
require "minitest/autorun"
require "ostruct"

def nodes_list
  {
    "nodes" => [
      { "scan_data" => OpenStruct.new(id: "abcd", status: "good", end_time: "a long time ago") },
      { "scan_data" => OpenStruct.new(id: "12345", status: "bad", end_time: "") },
    ]
  }
end

describe "nodes" do
  it "has an opaque error message" do
    nodes_list["nodes"].each { |node|
      assert_equal(node["scan_data"].id.length > 0, true)
      assert_equal(node["scan_data"].status.length > 0, true)
      assert_equal(node["scan_data"].end_time.length > 0, true)
    }
  end

  it "has a clearer error message" do
    assert nodes_list["nodes"].all? { |node|
      node["scan_data"].id.length > 0 &&
        node["scan_data"].status.length > 0 &&
        node["scan_data"].end_time.length > 0
    }, "Nodes did not have necessary scan_data: #{nodes_list}"
  end
end
```

Output:
```
Petes-MBP:tmp pete$ ruby test.rb
Run options: --seed 24770

# Running:

FF

Finished in 0.000998s, 2004.0087 runs/s, 7014.0303 assertions/s.

  1) Failure:
nodes#test_0001_has an opaque error message [test.rb:18]:
Expected: false
  Actual: true

  2) Failure:
nodes#test_0002_has a clearer error message [test.rb:23]:
Nodes did not have necessary scan_data: {"nodes"=>[{"scan_data"=>#<OpenStruct id="abcd", status="good", end_time="a long time ago">}, {"scan_data"=>#<OpenStruct id="12345", status="bad", end_time="">}]}

2 runs, 7 assertions, 2 failures, 0 errors, 0 skips
```

Reply:
oooh nice, thank you!